lib/src/data/chat.dart

```diff
@@ -0,0 +1,19 @@
+import "utils.dart";
+
+extension type ChatId(String value) {}
+
+class Chat {
+  List<String> messages;
+  Map<String, String> members;
+ 
+  Chat({});
+
+  Foo.fromJson(Json json) :
+	messages = json["messages"],
+	members = json["members"]
+
+  Json toJson() => {
+	"messages": messages,
+	"members": members
+  };
```

Comment:
```suggestion
  Chat.fromJson(Json json) :
	  messages = json["messages"],
	  members = json["members"]

  Json toJson() => {
	  "messages": messages,
	  "members": members
  };
```

Indentation and naming aside, remember that a list of anything that's not a simple string, number, bool, or null needs to be processed manually. For example: 
```dart
messages = [
  for (final messageJson in json["messages"])
    Message.fromJson(messageJson),
],

"messages": [
  for (final message in messages)
    message.toJson(),
],
```